### PR TITLE
Replace tasty with hspec

### PR DESCRIPTION
**Pull request by parsonsmatt**
All tasty tests have been converted to `hspec`.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+indent_style = tabs
+indent_size = 8
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.{hs,md,yml,yaml}]
+indent_style = space
+indent_size = 2
```
Comment (parsonsmatt) [inline .editorconfig:14]: Probably want to get an HLint and a Stylish Haskell config in here too

File: etc/CONTRIBUTORS [hunk 2]
```diff
@@ -43,3 +43,6 @@ james@earldouglas.com
 
 Eric Torreborre
 etorreborre@yahoo.com
+
+Matt Parsons
+parsonsmatt@gmail.com
```
Comment (parsonsmatt) [inline etc/CONTRIBUTORS:48]: join up folks

File: package.yaml [hunk 3]
```diff
@@ -16,15 +16,15 @@ copyright: ! 'Copyright (C) 2010-2013 Tony Morris
...
+homepage: https://github.com/bitemyapp/fp-course
+git: git@github.com:bitemyapp/fp-course.git
+bug-reports: https://github.com/bitemyapp/fp-course/issues
 
 tested-with: GHC==8.0.2
 
 extra-source-files:
 - etc/CONTRIBUTORS
-- changelog
+- CHANGELOG.md
```
Comment (parsonsmatt) [inline package.yaml:27]: typo i spose

**Review by parsonsmatt — COMMENTED** [review]
Net -290, nice, even with some more spacious formatting.

File: package.yaml (hunk 3 above)
Comment (bitemyapp) [reply package.yaml:27]: My bad, I renamed the file and forgot to edit that part.

**Pull request merged**